# Data Preprocessing for comments
listing_id             3831
date             2019-05-06
reviewer_id       224698181
comments                  .

comments liks this needs to be reomved. Also we are working on english corpus only

In [1]:
import pandas as pd
import numpy as np

In [2]:
input_df = pd.read_csv('Clean_reviews.csv')
print(len(input_df))

1273227


In [4]:

print(input_df.shape)
input_df.head()

(1273227, 4)


listing_id        date  reviewer_id  \
0        2060  2008-09-22         2865   
1        2595  2009-11-21        50679   
2        2595  2009-12-05        53267   
3        2595  2009-12-10        38960   
4        2595  2010-04-09        71130   

                                            comments  
0  very nice neighborhood,close enough to "A" tra...  
1  Notre séjour de trois nuits.\r\nNous avons app...  
2                                  Great experience.  
3  I've stayed with my friend at the Midtown Cast...  
4  We've been staying here for about 9 nights, en...

## Using LangDetect library 
Langdetect api does not parse the alphanumeric and numeric values, so we have to do exception handling as it throws errors when it encounters non-alpha characters.


In [47]:
filtered_list = []
from langdetect import detect
for i in range(len(input_df)):
    try:
        language = detect(input_df.iloc[i].comments)
        if language=='en':
            filtered_list.append(input_df.iloc[i])
    except:
        print("This row throws an error:", input_df.iloc[i].comments + str(i))

This row throws an error: .302
This row throws an error: .2629
This row throws an error: :)3657
This row throws an error: ..5755
This row throws an error: =(9199
This row throws an error: ✅11000
This row throws an error: .14313
This row throws an error: .14316
This row throws an error: .14369
This row throws an error: .14580
This row throws an error: .17240
This row throws an error: ..18277
This row throws an error: .19196
This row throws an error: 419244
This row throws an error: 120988
This row throws an error: .21266
This row throws an error: .23081
This row throws an error: 12324192
This row throws an error: .24355
This row throws an error: _27402
This row throws an error: --28887
This row throws an error: .29473
This row throws an error: ...29976
This row throws an error: .31941
This row throws an error: .32014
This row throws an error: .37129
This row throws an error: .38788
This row throws an error: .40890
This row throws an error: .43224
This row throws an error: -45375
This ro

This row throws an error: 10/10302479
This row throws an error: .303463
This row throws an error: .304223
This row throws an error: -305902
This row throws an error: 10/10306006
This row throws an error: .306530
This row throws an error: -308983
This row throws an error: .309688
This row throws an error: .310246
This row throws an error: -311001
This row throws an error: .311478
This row throws an error: .311586
This row throws an error: .313006
This row throws an error: :)313335
This row throws an error: !314438
This row throws an error: '314561
This row throws an error: .314733
This row throws an error: -314941
This row throws an error: .316472
This row throws an error: .316481
This row throws an error: .316555
This row throws an error: .316560
This row throws an error: .319374
This row throws an error: .320471
This row throws an error: .321861
This row throws an error: ❣️322396
This row throws an error: .322945
This row throws an error: .323210
This row throws an error: .324544
This

### Creating a new dataframe with filtered list

In [50]:
filtered_df = pd.DataFrame(filtered_list,columns=['listing_id','date','reviewer_id','comments'])

In [51]:
filtered_df.head()

listing_id        date  reviewer_id  \
0        2060  2008-09-22         2865   
3        2595  2009-12-10        38960   
4        2595  2010-04-09        71130   
5        2595  2010-05-25       117113   
6        2595  2012-05-07      1783688   

                                            comments  
0  very nice neighborhood,close enough to "A" tra...  
3  I've stayed with my friend at the Midtown Cast...  
4  We've been staying here for about 9 nights, en...  
5  We had a wonderful stay at Jennifer's charming...  
6  Hi to everyone!\r\nWould say our greatest comp...

In [52]:
filtered_df.to_csv('preprocessed_reviews.csv',index=False)

<b>Out of 1273227, 153889 comments are removed based on non-english language and non-alpha chracters.<b>

In [78]:
preprocessed_file = pd.read_csv('preprocessed_reviews.csv')
print(len(preprocessed_file))

1119338


In [79]:
preprocessed_file.head()

listing_id        date  reviewer_id  \
0        2060  2008-09-22         2865   
1        2595  2009-12-10        38960   
2        2595  2010-04-09        71130   
3        2595  2010-05-25       117113   
4        2595  2012-05-07      1783688   

                                            comments  
0  very nice neighborhood,close enough to "A" tra...  
1  I've stayed with my friend at the Midtown Cast...  
2  We've been staying here for about 9 nights, en...  
3  We had a wonderful stay at Jennifer's charming...  
4  Hi to everyone!\r\nWould say our greatest comp...

(1119338, 4)